add graph reference section to expressions (#1162)
neo4j/docs-cypher: modules/ROOT/pages/clauses/use.adoc

[[query-use-syntax]]
== Syntax

[source, syntax, role="noheader"]
----
USE <graph reference>
<other clauses>
----

A graph reference can be described using:

* Direct graph references: `USE db1`.
* The graph function xref:functions/graph.adoc#functions-graph-by-elementid[`graph.byElementId()`], to access a graph of a given element: `USE graph.byName(<element-id-string>)`.

When connected to a composite database, a graph reference may additionally be passed with:

* The graph function xref:functions/graph.adoc#functions-graph-byname[`graph.byName()`], which allows the graph reference to be resolved dynamically: `USE graph.byName(<string-expression>)`.

A more detailed description of how and when a graph references needs to be quoted and/or escaped is defined xref::queries/expressions.adoc#graphreferences[here].

== USE clause when connected to a standard or system database

> All databases and aliases are valid graph reference targets except composite databases and their constituents.

Reviewer: ```suggestion
All databases and aliases are valid graph reference targets except link:operations-manual/current/database-administration/aliases/manage-aliases-composite-databases/[composite databases and their constituents].
```

[[query-use-syntax]]
== Syntax

[source, syntax, role="noheader"]
----
USE <graph reference>
<other clauses>
----

A graph reference can be described using:

* direct graph references: `USE db1`

> * the graph function xref:functions/graph.adoc#functions-graph-by-elementid[`graph.byElementId()`], to access a graph of a given element: `USE graph.byName(<element-id-string>)`

Reviewer: ```suggestion
* Direct graph references: `USE db1`.
* The graph function xref:functions/graph.adoc#functions-graph-by-elementid[`graph.byElementId()`], to access a graph of a given element: `USE graph.byName(<element-id-string>)`.
```

When connected to a composite database, a graph reference may additionally be passed with:

> * the graph function xref:functions/graph.adoc#functions-graph-byname[`graph.byName()`], which allows the graph reference to be resolved dynamically: `USE graph.byName(<string-expression>)`

Reviewer: ```suggestion
* The graph function xref:functions/graph.adoc#functions-graph-byname[`graph.byName()`], which allows the graph reference to be resolved dynamically: `USE graph.byName(<string-expression>)`.
```

A more detailed description of how and when a graph references needs to be quoted and/or escaped is defined xref::queries/expressions.adoc#graphreferences[here].

== USE clause when connected to a standard or system database

> === Allowed graph reference targets

Reviewer: I don't think this heading is necessary
```suggestion
```

> All databases except composite databases can be targeted. It is not possible to target more than one database. This is only possible when connected to a composite database.

Reviewer: If possible, stick to 1 line per sentence.
```suggestion
All databases are valid graph reference targets except composite databases.
Targeting multiple databases is not allowed, unless connected to a composite database.
```

=== Position of use clauses
When connected to a non-composite database, the `USE` clause can only appear as the prefix of schema commands, or as the first clause of queries. There may be multiple `USE` clauses as long as they target the same database.

== USE clause when connected to a composite database

> === Allowed graph reference targets

Reviewer: ```suggestion
```

> When executing queries against a composite database, the `USE` clause must only refer to graphs that are part of the current composite database. The constituents can be listed either with `RETURN graph.names()` when connected to the composite database or `SHOW DATABASES YIELD name, constituents RETURN *`.

Reviewer: ```suggestion
When executing queries against a link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/concepts/[composite database], the `USE` clause must only refer to graphs that are part of the current composite database. The constituents can be listed either with `RETURN graph.names()` when connected to the composite database or `SHOW DATABASES YIELD name, constituents RETURN *`.
```

=== Position of use clauses

> When running queries against a link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/concepts/[composite database], the `USE` clause can appear as the first clause of:

Reviewer: ```suggestion
When running queries against a composite database, the `USE` clause can appear as the first clause of:
```

> * a query (as when connected to a non-composite database)

Reviewer: I stumbled on the "as when" part, so elaborated it slightly.
```suggestion
* A query (similar to how it is used when connected to a non-composite database).
```
